File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
    vector<bool> vis;
    vector<vector<int>> ans;
    vector<int> curr;

    void f(vector<int> &nums) {
        if(curr.size() == nums.size()) {
            ans.push_back(curr);
            return;
        }

        for(int j = 0; j < nums.size(); j++) {
            if(!vis[j]) {
                vis[j] = true;
                curr.push_back(nums[j]);
                f(nums);
                curr.pop_back();
                vis[j] = false;
            }
        }
    }
    vector<vector<int>> permute(vector<int>& nums) {
        vis.resize(100, false);
        f(nums);
        return ans;
    }
};

File: Data Structures & Algorithms/counting-bits/submission-0.cs
public class Solution {
    public int count(int n) {
        int ans = 0;
        for(int i = 0; i < 32; i++) {
            if((n & (1 << i)) != 0) {
                ans++;
            }
        }
        return ans;
    }
    public int[] CountBits(int n) {
        int[] ans = new int [n+1];
        for(int i = 0; i <= n;i++) {
            ans[i] = count(i);
        }
        return ans;
    }
}
